cmd/containerd-shim-runhcs-v1/pod.go

```diff
@@ -141,35 +141,52 @@ func createPod(ctx context.Context, events publisher, req *task.CreateTaskReques
 		id:     req.ID,
 		host:   parent,
 	}
-	// TOOD: JTERRY75 - There is a bug in the compartment activation for Windows
+
+	var netSetup hcsoci.NetworkSetup
+	if parent != nil {
+		if parent.NCProxyEnabled() {
+			caAddr := fmt.Sprintf(podComputeAgentAddrFmt, req.ID)
```

Comment:
As stated in my other comment, it seems like all the code in this conditional should move into a `NewExternalNetworkSetup` function, rather than being here.

Reply:
Done (as well as added NewInternalNetworkSetup)